### ISSUE TITLE
Wabbajack crashes reassembling Fallout 4 BA2 file on list install.

### ISSUE THREAD

**Issue by Yagisan** [wabbajack-tools/wabbajack#2249]
**Describe the bug**

During mod list installation, Wabbajack 3.0.4.1 crashes when attempting to reassemble a Fallout 4 BA2 file.  The log reports "End of stream before end of limit"

**Additional Context**

OS: Windows 10 Home. Version 21H2 (OS Build 19044.2130)
Game : Fallout 4 1.10.163.0

**Wabbajack Version**

3.0.4.1

**Logs**

[Wabbajack.current.log](https://github.com/wabbajack-tools/wabbajack/files/10263600/Wabbajack.current.log)

**Comment by Yagisan** [comment]
For the sake of thoroughness, I stepped back through previous version of Wabbajack that I have. I recreated the wabbajack file and tried installing it with each of the following versions.
2.5.3.28
3.0.3.1
3.0.4.0

3.0.3.1 and 3.0.4.0 failed exactly the same way as 3.0.4.1 - **however 2.5.3.28 correctly built, and correctly installed**. 3.0.4.1 was unable to successfully install the wabbajack file created by 2.5.3.28

**Comment by Yagisan** [comment]
I posted this in Discord, but for the record, here is a link to the wabbajack file that reproduces this crash. [https://www.dropbox.com/s/20wl7645f9zfaph/Rebuilding%20the%20Commonwealth.wabbajack?dl=0](https://www.dropbox.com/s/20wl7645f9zfaph/Rebuilding%20the%20Commonwealth.wabbajack?dl=0)

**Comment by Yagisan** [comment]
Just in case it gets lost in the discord scrollback

> halgari — Today at 06:31
> @GingasVR , @Yagisan dumping this here so I don't forget next week when I figure out how to fix it:
> 
> The issue happens when we have a source file being recompressed (at a texture level) and being included into a BA2 archive. When including the file the BA2 creator is trying to use the original size/format instead of the recompressed size and format. I'm not sure why this is happening, and what has changed, so I'm going to have to dig into the changes in this code since 2.5 to figure out what's going on

**Comment by Yagisan** [comment]
Replicated on commit a413725

[Wabbajack.current.log](https://github.com/wabbajack-tools/wabbajack/files/10336893/Wabbajack.current.log)

Direct Link to the ba2 file it can't reassemble https://mega.nz/file/19RGiYbC#IITKK7jbsucg6-hiFHILZ5o0g2JQck9WG8Sk-7MqwHI

Link to .wabbajack file used for installation https://mega.nz/file/NxwgjKRK#N1uzyhlSmbClrnT05T7OEiiC3-ahGwI8TAwUfgcb0t0

**Comment by Yagisan** [comment]
Just in case it gets lost in the discord scrollback

> halgari — 29/12/2022 03:06
> Attachment file type: archive
> [3.0.5.1.zip](https://cdn.discordapp.com/attachments/640958797820461082/1057721321720660038/3.0.5.1.zip)
> 123.33 MB
> @Yagisan here's a new build for you to try, delete/raname the GlobaVFSCache3.sqlite in AppData/Local/Wabbajack then recompile your fo4 list and try to install it,
> or even better upload the .wabbajack here and I'll try to install it as well
> Basically until this week we had no support in WJ for cubemaps

For the record, it did not work.

**Comment by Yagisan** [comment]
I have replicated this bug with Wabbajack 3.0.6.0

`GlobalVFSCache3.sqlite`, and for good measure, `GlobalHashCache.sqlite`, and the entire `PatchCache` folder were deleted.

Wabbajack 3.0.6.0 was used to build the Wabbajack. It was then closed, reopened and used to install. It failed with the now familiar ` End of stream before end of limit`

Please find attached the log.
[Wabbajack.current.log](https://github.com/wabbajack-tools/wabbajack/files/10474343/Wabbajack.current.log)

The direct Link to the ba2 file it can't reassemble remains [https://mega.nz/file/19RGiYbC#IITKK7jbsucg6-hiFHILZ5o0g2JQck9WG8Sk-7MqwHI
](https://mega.nz/file/19RGiYbC#IITKK7jbsucg6-hiFHILZ5o0g2JQck9WG8Sk-7MqwHI
)

Link to .wabbajack file used for installation [https://mega.nz/file/19AFAY7Y#3KOrnatcq7v8aOcIaslCW_AfjO8QHij_5etg7jczYFo](https://mega.nz/file/19AFAY7Y#3KOrnatcq7v8aOcIaslCW_AfjO8QHij_5etg7jczYFo)

**Comment by Yagisan** [comment]
I am afflicted with the Fallout 4 ba2 bug.  My system is somehow at the system dependent ba2 limit. This is a very frustrating issue, that is unlikely to ever be fixed by Bethesda. So, there are two common solutions to this. The first is to unpack all ba2 files, which is not great. The second is to repack as many ba2 files as possible into less ba2 files. This is the route I have taken.

Previous testing indicated that Wabbajack was having problems processing cubemaps while reassembling ba2 files. Unfortunately for me Wabbajack 3.0.6.0 was unable to install the repacked ba2 files.

I decided to do further testing. I unpacked my selected ba2 files from the original downloads, taking care to exclude cubemaps. I then repacked the textures into a ba2 file, without altering the textures in any way. 
I then installed it, and compiled the wabbajack file. The compilation log is attached.
[Wabbajack.compile.log](https://github.com/wabbajack-tools/wabbajack/files/10475322/Wabbajack.compile.log)

Over the course of multiple test installs, I noticed an interesting log message. `Recompressing` followed by a texture name.  As the textures should be unchanged, it leads me to question, why is Wabbajack recompressing these textures, and given the `End of stream before end of limit` error references a texture, I start to wonder - is this recompression process breaking the textures? Attached is a failed installation log 
[Wabbajack.current.log](https://github.com/wabbajack-tools/wabbajack/files/10475323/Wabbajack.current.log)
